### PR TITLE
Add reputation server integration

### PR DESCRIPTION
**Pull request by nvonpentz**
When a browser claims a grant, the grant server should send the paymentId to the reputation server and ask if there is sufficient reputation to claim a grant.  If there is, continue with the grant claim.

This PR adds the reputation client for communication with reputation server and uses it during the grant claim event.

### CODE REVIEW

File: utils/reputation/client.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	return resp, fmt.Errorf("Request error: %d", resp.StatusCode)
+}
+
+// IsWalletReputableResponse is what the reputation server
+// will send back when we ask if a wallet is reputable
+type IsWalletReputableResponse struct {
+	IsReputable bool `json:"isReputable"`
+}
+
+// IsWalletReputable makes the request to the repuation server
```
Comment (evq) [inline utils/reputation/client.go]: reputation

File: utils/reputation/client.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	BaseURL   *url.URL
+	AuthToken string
+
+	client *http.Client
+}
+
+// New returns a new HTTPClient, retrieving the base URL from the
+// environment
+func New() (*HTTPClient, error) {
+	serverURL := os.Getenv("REPUATION_SERVER")
```
Comment (evq) [inline utils/reputation/client.go]: REPUTATION_SERVER

File: promotion/claim.go [hunk 3]
```diff
@@ -99,8 +104,14 @@ func (service *Service) ClaimPromotionForWallet(ctx context.Context, promotionID
 	}
 
 	// TODO lookup and return existing claim if exists?
```
Comment (nvonpentz) [inline promotion/claim.go]: @evq Should I resolve this TODO / do you know what I need to do?

Comment (evq) [reply promotion/claim.go]: ah! thanks for raising this - opened https://github.com/brave-intl/bat-go/issues/118. let's address in a separate PR

**Pull request merged**